mempool/mempool.go

```diff
@@ -308,8 +329,11 @@ func (mem *Mempool) CheckTx(tx types.Tx, cb func(*abci.Response)) (err error) {
 	// use defer to unlock mutex because application (*local client*) might panic
 	defer mem.proxyMtx.Unlock()
 
-	if mem.Size() >= mem.config.Size {
-		return ErrMempoolIsFull
+	if mem.Size() >= mem.config.Size ||
+		int64(len(tx))+mem.TxsTotalBytes() > mem.config.MaxTxsTotalBytes {
+		return ErrMempoolIsFull{
+			mem.Size(), mem.config.Size,
+			mem.TxsTotalBytes(), mem.config.MaxTxsTotalBytes}
```

Comment:
Isn't it possible for TxsTotalBytes() to change between when we checked and when we construct the error? 

We should probably store it in a local var and use that to avoid the potential (though relatively insignificant) race condition!